### PR TITLE
kafka: shade in client jar so we can include other client versions elsewhere

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

**Comment by pennello** [comment]
There's no actual reason to deploy snapshot builds of components for CI testing.

screen shot 2017-02-01 at 8 25 00 am

http://teamcity.otenv.com/viewLog.html?buildId=5854003&buildTypeId=ArchitectureTeam_JavaPlatform_OtjLogging_BuildConfig&tab=buildLog#_focus=629

@stevenschlansker What do you think about changing the config for this to just run a `mvn verify`.  Will that still do the right thing for multi-module builds with sub-modules that depend on other modules?

File: kafka/pom.xml [hunk 1]
```diff
@@ -31,6 +31,38 @@
...
+          <relocation>
+            <pattern>org.apache.kafka</pattern>
+            <shadedPattern>otj.logging.kafka</shadedPattern>
+          </relocation>
+        </relocations>
+      </configuration>
+    </plugin>
+  </plugins>
+  </build>
+
```
Comment (pennello) [inline kafka/pom.xml]: So this means we'll have to be more on top of updates for this library if we want to include, for example, bugfixes in upstream Kafka.  How will we make sure we're on top of these updates?

**Comment by stevenschlansker** [comment]
No, you would at least need to do a `mvn install`.  `verify` is not sufficient.

File: kafka/pom.xml (hunk 1 above)
Comment (stevenschlansker) [inline kafka/pom.xml]: Dunno.  We will just have to stay on top of it.

Comment (pennello) [inline kafka/pom.xml]: We need to do better than that.

At least follow this feed, perhaps.
https://github.com/apache/kafka/releases.atom

Are you on their mailing list?

**Comment by pennello** [comment]
@stevenschlansker 

> No, you would at least need to do a `mvn install`.  `verify` is not sufficient.

That's a good idea, though.  Let me give that a try.

**Comment by pennello** [comment]
> Let me give that a try.

[Nice!][1]

[1]: https://github.com/opentable/otj-logging/commit/a46cf8a01d0ae2b042907db32fdf273c70ba816c#commitcomment-20700747

**Comment by pennello** [comment]
Travis is still failing due to this nonsense.

    [ERROR] Plugin com.ning.maven.plugins:maven-dependency-versions-check-plugin:2.0.2-OT-2 or one of its dependencies could not be resolved: Failure to find com.ning.maven.plugins:maven-dependency-versions-check-plugin:jar:2.0.2-OT-2 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]

**Comment by stevenschlansker** [comment]
Now that we are off the Spring betas, we may not need that patch as desperately.  We should try reverting.

**Comment by stevenschlansker** [comment]
New Kafka libraries support cross-release compatibility, so this shouldn't be needed thankfully

**Comment by pennello** [comment]
Nice.

**Pull request closed without merging**